### PR TITLE
sdk-format autopep code formatter

### PR DESCRIPTION
**Pull request by barchen1**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/24189
fixes: https://github.com/demisto/etc/issues/25177

## Description
Add a code formatter named "black" for python files.

## Screenshots
Screen Shot 2020-06-02 at 11 14 42

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Changelog
+* Add a code formatter for python files
 * Add support for new classifiers and mappers
```
Comment (yaakovi) [inline CHANGELOG.md:3]: those should be in the past tense 
```suggestion
* Added a code formatter for python files.
* Added support for new classifiers and mappers.
```

File: demisto_sdk/commands/format/format_module.py [hunk 2]
```diff
@@ -32,6 +33,7 @@
     'layout': LayoutJSONFormat,
     'dashboard': DashboardJSONFormat,
     'classifier': ClassifierJSONFormat,
+    'pythonfile': PythonFileFormat
```
Comment (yaakovi) [inline demisto_sdk/commands/format/format_module.py:36]:
```suggestion
    'pythonfile': PythonFileFormat,
```

File: demisto_sdk/commands/format/format_module.py [hunk 3]
```diff
@@ -47,11 +49,11 @@ def format_manager(input: str = None, output: str = None, from_version: str = No
         int 0 in case of success 1 otherwise
     """
     if input:
-        files = get_files_in_dir(input, ['json', 'yml'])
+        files = get_files_in_dir(input, ['json', 'yml', 'py'])
     else:
         files = [file['name'] for file in
                  get_changed_files(filter_results=lambda _file: not _file.pop('status') == 'D')]
-    if output and not output.endswith(('.yml', 'json')):
+    if output and not output.endswith(('.yml', 'json', 'py')):
```
Comment (yaakovi) [inline demisto_sdk/commands/format/format_module.py:56]: why `yml` has a dot and the other two don't?

File: demisto_sdk/commands/format/format_module.py [hunk 4]
```diff
@@ -115,6 +116,8 @@ def logger(input: str, format_res: int, validate_res: int) -> Tuple[List[str], L
         if validate_res == 2:
             error_list.append(f'Format Status   on file: {input} - Failed')
             skipped_list.append(f'Validate Status on file: {input} - Skipped')
+        elif validate_res == 3:
```
Comment (yaakovi) [inline demisto_sdk/commands/format/format_module.py:119]: use the constants and not magic numbers

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        Attributes:
+            input (str): the path to the file we are updating at the moment.
+            output (str): the desired file name to save the updated version of the python file to.
+    """
+
+    def __init__(self, input: str = '', output: str = '', path: str = '', from_version: str = '',
+                 no_validate: bool = False):
+        super().__init__(input, output, path, from_version, no_validate)
+        self.no_validate = True
```
Comment (yaakovi) [inline demisto_sdk/commands/format/update_pythonfile.py:23]: `no_validate` has a default value of `False`, which you pass to the base class and then override it.
this is confusing, add a comment on why you force it to be `True`.
or change the default to `True` and remove the override.

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            bool. True if succeed to run black on file, False otherwise.
+        """
+        print("\nRunning black on file: {}\n".format(py_file_path))
+        try:
+            if subprocess.call(["black", "--skip-string-normalization", "-v", "--line-length", "120", py_file_path])\
+                    == BLACK_INTERNAL_ERROR:
+                return False
+            return True
+        except FileNotFoundError:
+            return "black skipped! It doesn't seem you have black installed.\n " \
```
Comment (yaakovi) [inline demisto_sdk/commands/format/update_pythonfile.py:39]: the function should return a bool, not a string.
this should be printed as error and return `False`

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            input (str): the path to the file we are updating at the moment.
+            output (str): the desired file name to save the updated version of the python file to.
+    """
+
+    def __init__(self, input: str = '', output: str = '', path: str = '', from_version: str = '',
+                 no_validate: bool = False):
+        super().__init__(input, output, path, from_version, no_validate)
+        self.no_validate = True
+
+    def is_format_by_black(self, py_file_path):
```
Comment (yaakovi) [inline demisto_sdk/commands/format/update_pythonfile.py:25]: please update the function name.
functions called `is_*` usually checks conditions and doesn't modify data.

File: CHANGELOG.md (hunk 1 above)
Comment (barchen1) [reply CHANGELOG.md]: done

File: demisto_sdk/commands/format/format_module.py (hunk 2 above)
Comment (barchen1) [reply demisto_sdk/commands/format/format_module.py]: done

File: demisto_sdk/commands/format/format_module.py (hunk 3 above)
Comment (barchen1) [reply demisto_sdk/commands/format/format_module.py]: no reason,
changed.

File: demisto_sdk/commands/format/format_module.py (hunk 4 above)
Comment (barchen1) [reply demisto_sdk/commands/format/format_module.py]: done

File: demisto_sdk/commands/format/update_pythonfile.py (hunk 5 above)
Comment (barchen1) [reply demisto_sdk/commands/format/update_pythonfile.py]: done

File: demisto_sdk/commands/format/update_pythonfile.py (hunk 7 above)
Comment (barchen1) [reply demisto_sdk/commands/format/update_pythonfile.py]: done

File: demisto_sdk/commands/format/update_pythonfile.py (hunk 6 above)
Comment (barchen1) [reply demisto_sdk/commands/format/update_pythonfile.py]: done

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    def format_py_using_autopep(self, py_file_path):
+        """Run autopep8 formatter on python file.
+        Args:
+            py_file_path (str): The python file path.
+        Returns:
+            bool. True if succeed to run autopep8 on file, False otherwise.
+        """
+        print("\nRunning autopep8 on file: {}\n".format(py_file_path))
+        try:
+            if subprocess.call(["autopep8", "-i", "--max-line-length", "130", py_file_path]) \
```
Comment (ronykoz) [inline demisto_sdk/commands/format/update_pythonfile.py:34]: why 130? i think it should be 120 + let's make it a constant

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+                    == BLACK_INTERNAL_ERROR:
+                return False
+            return True
+        except FileNotFoundError:
```
Comment (ronykoz) [inline demisto_sdk/commands/format/update_pythonfile.py:38]: maybe the python file doesn't exist rather then autopep8, no?

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        is_autopep_passed = self.format_py_using_autopep(str(self.source_file))
+        if is_autopep_passed:
+            print_color(F'=======Finished updates for files: {self.output_file}=======\n', LOG_COLORS.WHITE)
+            return SUCCESS_RETURN_CODE
+        return ERROR_RETURN_CODE
+
+    def format_file(self) -> Tuple[int, int]:
+        """Manager function for the integration python updater."""
+        format = self.run_format()
+        return format, SKIP_VALIDATE_PY_RETURN_CODE
```
Comment (ronykoz) [inline demisto_sdk/commands/format/update_pythonfile.py:56]: SKIP_VALIDATE_PY_RETURN_CODE what's it used for?

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+from demisto_sdk.commands.format.update_generic_yml import BaseUpdate
+
+BLACK_INTERNAL_ERROR = 123
+
+
+class PythonFileFormat(BaseUpdate):
+    """PythonFileFormat class is designed to update python file according to Demisto's convention.
+
+        Attributes:
+            input (str): the path to the file we are updating at the moment.
```
[the remark this answers is not in the document]
Comment (ronykoz) [reply demisto_sdk/commands/format/update_pythonfile.py:16]: @barchen1 update this one please

File: demisto_sdk/commands/format/update_pythonfile.py [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+                        "Then run: autopep8 -i {}".format(py_file_path), LOG_COLORS.YELLOW)
+            return False
+        return True
+
+    def create_output_file(self):
+        """Create output file with the data of the source file."""
+        with open(str(self.source_file), 'r') as source_file:
+            file_data = source_file.readlines()
+        with open(self.output_file, 'w') as output_file:
+            output_file.writelines(file_data)
```
Comment (yaakovi) [inline demisto_sdk/commands/format/update_pythonfile.py:49]: why not use `shutil.copy`?

Comment (barchen1) [reply demisto_sdk/commands/format/update_pythonfile.py]: changed

**Pull request merged**